### DISCUSSION TITLE
Bed tramming inset in Manual Mesh firmware configuration

### DISCUSSION THREAD

**Discussion by lookalek** [mriscoc/Ender3V2S1#724] [Q&A] [answered]
Hi guys,

I am getting rid of the CRTouch and swapping it to fully manual mesh leveling with a Z switch installed.
I am using Ender3 S1 which bed size is 235x235 mm, but still sticking to the usable printing area of 220x220, what leaves a margin of 15 mm all around.

To avoid clipping during meshing, by default in the firmware an offset of 10 mm is implemented (_#define PROBING_MARGIN 10_). But this offset is from the 220 mm, not from the 235 mm, so I changed it to 0 mm what now gives me a possibility to do the manual meshing from the very edges of the 220x220 mm area. Perfect, this one went smooth.

I am having a problem with the bed tramming wizard though. The offset from edges is set to 30 mm, and I have no idea how to change it, so it equals to 0 mm.
I tried to change  _#define BED_TRAMMING_INSET_LFRB { 30, 30, 30, 30 }_ and _#define TRAMMING_POINT_XY { { 29, 29 }, { 199, 29 }, { 199, 199 }, { 29, 199 } }_ with no success.

Does anyone know how to do it in configuration files?

Thanks!

**Comment by brew99** [comment]
did you #define LCD_BED_TRAMMING in config.h? (just above #define BED_TRAMMING_INSET_LFRB { 30, 30, 30, 30 })

**Comment by mriscoc** [comment, marked as answer]
Bed Tramming margin is defined here:
https://github.com/mriscoc/Ender3V2S1/blob/68791f246b8da6ff3e69b2a88b601d36fd73e9e7/Marlin/src/lcd/e3v2/proui/dwin.cpp#L2388-L2397

**Reply by lookalek** [reply]
Great, thank you. Works like a charm now.

**Comment by BarryHendriks** [comment]
My BIQU-B1 3D printer has the following cards: MotherBoard: BTT SKR V1.4 with CPU  LPC1768 and Screen Card: TFT35_B1_V3.0 with a bed size of 235mm x 235mm x 270mm.

I am trying to install a BIQU Microprobe V2 (Oposite logic to BL Touch) to a standard (No modifications) BIQU-B1 3D printer. The printer has a dedicated factory build socket on-board already wired to the BL Touch port at the back to install the BL Touch. No need for wiring or other changes except for configuration changes in firmware.

I am already busy for weeks to try and make the BIQU Microprobe V2 work by resolving any and all issues the Platform IO reports while building the Marlin firmware.bin file.

My probe is installed in front of the print head and I defined the following: NOZZLE_TO_PROBE_OFFSET ( 0, -36, -1,7 ) 
 //Probe in line with nozzle (X = 0). Measured 36mm from nozzle to probe tip. Probe in stow / retracted position 1.7mm higher than nozzle.
and PROBE_MARGIN 9 //To keep nozzle on the bed but maximize bed usable print area.

One of the issues that I just can't resolve is the LCD_BED_TRAMMING

Just to clarify - maybe I am understanding BED_TRAMMING_INSET_LFRB wrong. My understanding is that Left is at 0 and Front is at 0. Right is the size of the bed accross (235mm) and back is the size of the bed from front to back (235mm) - you only need to subtract the PROBE_MARGIN and if you have a probe installed the NOZZLE_TO_PROBE_OFFSET subtract from positions installed.

I have changed from inserting values/numbers to a more programmatic approach but neither gets accepted by Platform IO firmware build.
The only values that do pass Platform IO firmware build is after hundreds of times changing values - safe the files and build firmware until it worked / pass without fail:  BED_TRAMMING_INSET_LFRB { 117, 81, 9, 60 }  BUT this is definitely not correct and does not make sense.

Even the default value for BED_TRAMMING_INSET_LFRB before any changes suggest a low value that is the same for all 4 settings: BED_TRAMMING_INSET_LFRB { 30, 30, 30, 30 } - Thus instead of all different values I got it should have been simple: { 9, (9+36), 9, 9}

This programmatic approach makes sense and SHOULD be right but it fails Platform IO firmware build stating that the back is not reachable.
BED_TRAMMING_INSET_LFRB { PROBING_MARGIN, PROBING_MARGIN + 36, X_BED_SIZE - PROBING_MARGIN, Y_BED_SIZE - (PROBING_MARGIN + 36) }

Just wondering if this is perhaps a bug or error in Marlin ? Because the above should have worked without fail.  Maybe I defined something I should not have. What other settings would have an impact on LCD_BED_TRAMMING? I have searched but could not find any connection.

**Reply by mriscoc** [reply]
You can use the firmware as it is and use the physical settings menu to change the parameters, then after getting the correct values, put them in a G-code configuration file.
https://github.com/mriscoc/Ender3V2S1/wiki/Configuration-files
https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides

**Reply by BarryHendriks** [reply]
1st: Thank you for the prompt respose!

Q1: Does this mean I should comment out #define LCD_BED_TRAMMING or keep it enabled?
Q2: If stays enabled then still use my non-sense values that passed Platform IO build?

Sorry for the questions - first time I had to delve into firmware files to get something working because supplier not interested to configure self.

**Reply by mriscoc** [reply]
I think that your hardware is not compatible/tested with this firmware, you may need to use a pure Marlin firmware from https://github.com/MarlinFirmware/Marlin/releases
